js/foam/apps/builder/controller/Controller.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * @license
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+CLASS({
+  package: 'foam.apps.builder.controller',
+  name: 'Controller',
+
+  requires: [
+    'foam.apps.builder.AppConfig',
+    'foam.apps.builder.AppConfigCitationView',
+    'foam.apps.builder.controller.PermanentNav',
+    'foam.apps.builder.controller.StackView',
+    'foam.input.touch.GestureManager',
+    'foam.input.touch.TouchManager',
+    'foam.ui.DAOListView',
+    'foam.ui.md.PopupView',
+  ],
+  imports: [
+    // TODO(markdittmer): Just for testing purposes.
+    'setTimeout',
+  ],
+  exports: [
+    'stack',
+    'touchManager',
+    'gestureManager',
+  ],
+
+  properties: [
+    {
+      type: 'foam.input.touch.TouchManager',
```

Comment:
These are going to run afoul of the new `type` logic, aren't they? I mean, they should fail gracefully since there is no `foam.input.touch.TouchManagerProperty`, but that's still not ideal. I'd say it's better not to include `type`s like these.